### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by sivachandran**
Demo implementation for DB transaction abstraction in Go.

### CODE REVIEW

File: models/postgres/txcontext.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	err = tx.Commit()
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	return nil
+}
+
+type Tx interface {
```
Comment (sudarshan-reddy) [inline models/postgres/txcontext.go:39]: ~~Can we directly inherit sql.DB here?~~

**Review by sudarshan-reddy — COMMENTED** [review]
Looks great @sivachandran. Will adding tests further help readability?

File: models/postgres/txcontext.go (hunk 1 above)
Comment (sivachandran) [reply models/postgres/txcontext.go:39]: `DB` is a struct. We need an interface here.

**Comment by sudarshan-reddy** [comment]
@sivachandran can you merge this? Would like to use the repo as a citation for a demo. I can add unit tests in separate PR.

**Pull request merged**